<a href="https://colab.research.google.com/github/yashar1908/ML_MINOR_FEB/blob/main/ML_MINOR_FEB.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [22]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [23]:
data = pd.read_csv('https://raw.githubusercontent.com/ameer-aiml/ai-vodafone/main/datasets/car_data.csv')
data

Car_Name  Year  Selling_Price  Present_Price  Kms_Driven Fuel_Type  \
0       ritz  2014           3.35           5.59       27000    Petrol   
1        sx4  2013           4.75           9.54       43000    Diesel   
2       ciaz  2017           7.25           9.85        6900    Petrol   
3    wagon r  2011           2.85           4.15        5200    Petrol   
4      swift  2014           4.60           6.87       42450    Diesel   
..       ...   ...            ...            ...         ...       ...   
296     city  2016           9.50          11.60       33988    Diesel   
297     brio  2015           4.00           5.90       60000    Petrol   
298     city  2009           3.35          11.00       87934    Petrol   
299     city  2017          11.50          12.50        9000    Diesel   
300     brio  2016           5.30           5.90        5464    Petrol   

    Seller_Type Transmission  Owner  
0        Dealer       Manual      0  
1        Dealer       Manual      0  
2        Dealer       Manual      0  
3        Dealer       Manual      0  
4        Dealer       Manual      0  
..          ...          ...    ...  
296      Dealer       Manual      0  
297      Dealer       Manual      0  
298      Dealer       Manual      0  
299      Dealer       Manual      0  
300      Dealer       Manual      0  

[301 rows x 9 columns]

In [24]:
data.describe()

Year  Selling_Price  Present_Price     Kms_Driven       Owner
count   301.000000     301.000000     301.000000     301.000000  301.000000
mean   2013.627907       4.661296       7.628472   36947.205980    0.043189
std       2.891554       5.082812       8.644115   38886.883882    0.247915
min    2003.000000       0.100000       0.320000     500.000000    0.000000
25%    2012.000000       0.900000       1.200000   15000.000000    0.000000
50%    2014.000000       3.600000       6.400000   32000.000000    0.000000
75%    2016.000000       6.000000       9.900000   48767.000000    0.000000
max    2018.000000      35.000000      92.600000  500000.000000    3.000000

In the description it is shown that the minimum year in the data is 2003 and the maximum year is 2018. Therefore data from 2003 - 2018 is present.

In the description it is shown that the minimum price at which a vehicle was sold is 0.1

According to the description, the maximum price at which a vehicle was sold is 35.

In [25]:
data.shape

(301, 9)

The shape of the data matrix tells us that there is data about 301 cars present in the dataset.

In [26]:
data.isnull().sum().sum()

0

Thus there is no missing data in the dataset

In [27]:
data.nunique()

Car_Name          98
Year              16
Selling_Price    156
Present_Price    147
Kms_Driven       206
Fuel_Type          3
Seller_Type        2
Transmission       2
Owner              3
dtype: int64

As obtained, there are 98 different vehicles in the data. 

In [28]:
print(data['Fuel_Type'].value_counts()['CNG'])

2


The data includes 2 CNG vehicles

In [29]:
print(data['Seller_Type'].value_counts()['Individual'])

106


106 out of the 301 cars are thus for sale by individual sellers directly

In [30]:
'Auto' in data['Transmission'].values

False

There are no Auto Transmission cars in the dataset

In [31]:
print(data['Owner'].value_counts()[1])

10


Thus there are 10 cars which have 1 owner.

In [32]:
depreciation = data['Present_Price'] - data['Selling_Price']
depreciation.describe()

count    301.000000
mean       2.967176
std        4.828771
min        0.030000
25%        0.390000
50%        1.640000
75%        3.350000
max       57.600000
dtype: float64